pkg/cmd/clientConfig.go

```diff
@@ -77,35 +73,15 @@ kubectl plugin mobile get clientconfig`,
 				return err
 			}
 
-			ms := listServices(ns, ccc.k8Client)
-			for _, svc := range ms {
-				var svcConfig *ServiceConfig
-				var err error
-				includedService := true
-				configMap, err := ccc.k8Client.CoreV1().ConfigMaps(ns).Get(svc.Name, v1.GetOptions{})
-				// ignoring not found as some services will not have this public configmap
+			filter := v1.ListOptions{LabelSelector: fmt.Sprintf("mobileClientId=%s", clientID)}
+			secrets, err := ccc.k8Client.CoreV1().Secrets(ns).List(filter)
+			for _, secret := range secrets.Items {
+				convertor := defaultSecretConvertor{}
+				svcConfig, err := convertor.Convert(secret)
 				if err != nil {
-					if kerror.IsNotFound(err) {
-						continue
-					}
-					return errors.Wrap(err, "unable to create config. Failed to get service "+svc.Name+" configmap")
-				}
-				configType := configMap.ObjectMeta.Annotations["configType"]
-				if _, ok := convertors[svc.Name]; !ok {
-					convertor := defaultSecretConvertor{}
-					if svcConfig, err = convertor.Convert(svc.ID, configMap.Data, configType); err != nil {
-						return err
-					}
-				} else {
-					// we can only convert what is available
-					convertor := convertors[svc.Name]
-					if svcConfig, err = convertor.Convert(svc.ID, configMap.Data, configType); err != nil {
-						return err
-					}
-				}
-				if includedService {
-					ret = append(ret, svcConfig)
+					return err
```

Comment:
`return errors.Wrap(err, "unable to convert secret: " + secret.Name)`

Reply:
We are already wrapping the error in the convert method and returning the fact that we cant unmarsahl the config so not so sure that it is better bubble up an error with less info in it? wdyt?

Reply:
yeah, this makes sense to not wrap this as it's already wrapped [here](https://github.com/PhilipGough/mobile-cli/blob/8f81b34d97d335fac063fc30dcf379b6be05c9ea/pkg/cmd/types.go#L123).